### ISSUE TITLE
[Python 3] Fix unit.test_pyobjects.MapTests.test_map

### ISSUE THREAD

**Issue by Ch3LL** [saltstack/salt-jenkins#435]
The test `unit.test_pyobjects.MapTests.test_map` is failing on Fedora25

```
Error Message

Order[13 chars]ba', OrderedDict([('pkg.installed', [{'names':[104 chars]]))]) != Order[13 chars]ba', {'pkg.installed': [{'names': ['samba', 's[86 chars]]})])
Stacktrace

Traceback (most recent call last):
  File "/testing/tests/unit/test_pyobjects.py", line 426, in test_map
    assert_ret(ret, 'samba', 'samba-client', 'smbd')
  File "/testing/tests/unit/test_pyobjects.py", line 417, in assert_ret
    {'require': [{'pkg': 'samba'}]}
AssertionError: Order[13 chars]ba', OrderedDict([('pkg.installed', [{'names':[104 chars]]))]) != Order[13 chars]ba', {'pkg.installed': [{'names': ['samba', 's[86 chars]]})])
```

https://jenkins.saltstack.com/job/branch_tests/job/2017.7/view/Python%203/job/salt-linode-py3-fedora25/107/testReport/junit/unit.test_pyobjects/MapTests/test_map/

**Comment by terminalmage** [comment]
Fixed by https://github.com/saltstack/salt/pull/41951

**Comment by DmitryKuzmenko** [comment]
Thanks @terminalmage.
Anyway I've finished the work I've done for this and both could be used together without conflict. Just your `priority` feature will override the natural (declaration) priority that I've fixed in #41958.
